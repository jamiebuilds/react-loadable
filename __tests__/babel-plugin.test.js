// @flow
declare var test: any;
declare var expect: any;
import { transform } from "babel-core";
import plugin from "../src/babel";

let trim = str => {
  return str.replace(/\r\n?/g, "\n").trim();
};

let fn = (code, opts = {}) => {
  let input = trim(code);
  let options = { plugins: [[plugin, opts], "syntax-dynamic-import"] };
  return transform(input, options).code;
};

test("accept-default-options", () => {
  expect(
    fn(
      `
    import Loadable from "react-loadable";
    let LoadableMyComponent = Loadable({
      loader: () => import("./MyComponent"),
      LoadingComponent: MyLoadingComponent,
    });
  `
    )
  ).toMatchSnapshot();
});

test("accept-server-false-webpack-false", () => {
  expect(
    fn(
      `
    import Loadable from "react-loadable";
    let LoadableMyComponent = Loadable({
      loader: () => import("./MyComponent"),
      LoadingComponent: MyLoadingComponent,
    });
  `,
      {
        server: false,
        webpack: false
      }
    )
  ).toMatchSnapshot();
});

test("accept-server-false-webpack-true", () => {
  expect(
    fn(
      `
    import Loadable from "react-loadable";
    let LoadableMyComponent = Loadable({
      loader: () => import("./MyComponent"),
      LoadingComponent: MyLoadingComponent,
    });
  `,
      {
        server: false,
        webpack: true
      }
    )
  ).toMatchSnapshot();
});

test("accept-server-true-webpack-false", () => {
  expect(
    fn(
      `
    import Loadable from "react-loadable";
    let LoadableMyComponent = Loadable({
      loader: () => import("./MyComponent"),
      LoadingComponent: MyLoadingComponent,
    });
  `,
      {
        server: true,
        webpack: false
      }
    )
  ).toMatchSnapshot();
});

test("not-overwrite-existing-properties", () => {
  expect(
    fn(
      `
    import Loadable from "react-loadable";
    let LoadableMyComponent = Loadable({
      loader: () => import("./MyComponent"),
      LoadingComponent: MyLoadingComponent,
      serverSideRequirePath: null,
      webpackRequireWeakId: null
    });
  `,
      {
        server: true,
        webpack: true
      }
    )
  ).toMatchSnapshot();
});

test("track-imports-correctly", () => {
  expect(
    fn(
      `
    import L from 'react-loadable';
    import Loadable from 'not-react-loadable';
    let C1 = L({
      loader: () => import('./MyComponent')
    });
    let C2 = Loadable({
      loader: () => import('./MyComponent')
    });
  `,
      {
        server: true,
        webpack: true
      }
    )
  ).toMatchSnapshot();
});

test("add-webpack-chunk-name-property", () => {
  expect(
    fn(
      `
    import Loadable from "react-loadable";
    let LoadableMyComponent = Loadable({
      loader: () => import("./MyComponent" /* webpackChunkName: "my-component" */ ),
      LoadingComponent: MyLoadingComponent,
    });
  `,
      {
        server: false,
        webpack: true
      }
    )
  ).toMatchSnapshot();

  expect(
    fn(
      `
    import Loadable from "react-loadable";
    let LoadableMyComponent = Loadable({
      loader: () => import(/*webpackChunkName: "my-component"*/ "./MyComponent"),
      LoadingComponent: MyLoadingComponent,
    });
  `,
      {
        server: false,
        webpack: true
      }
    )
  ).toMatchSnapshot();
});
