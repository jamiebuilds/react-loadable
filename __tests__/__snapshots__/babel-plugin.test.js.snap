// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accept-default-options 1`] = `
"import _path from \\"path\\";
import Loadable from \\"react-loadable\\";
let LoadableMyComponent = Loadable({
  loader: () => import(\\"./MyComponent\\"),
  serverSideRequirePath: _path.join(__dirname, \\"./MyComponent\\"),
  LoadingComponent: MyLoadingComponent
});"
`;

exports[`accept-server-false-webpack-false 1`] = `
"import Loadable from \\"react-loadable\\";
let LoadableMyComponent = Loadable({
  loader: () => import(\\"./MyComponent\\"),
  LoadingComponent: MyLoadingComponent
});"
`;

exports[`accept-server-false-webpack-true 1`] = `
"import Loadable from \\"react-loadable\\";
let LoadableMyComponent = Loadable({
  loader: () => import(\\"./MyComponent\\"),
  webpackRequireWeakId: () => require.resolveWeak(\\"./MyComponent\\"),
  LoadingComponent: MyLoadingComponent
});"
`;

exports[`accept-server-true-webpack-false 1`] = `
"import _path from \\"path\\";
import Loadable from \\"react-loadable\\";
let LoadableMyComponent = Loadable({
  loader: () => import(\\"./MyComponent\\"),
  serverSideRequirePath: _path.join(__dirname, \\"./MyComponent\\"),
  LoadingComponent: MyLoadingComponent
});"
`;

exports[`add-webpack-chunk-name-property 1`] = `
"import Loadable from \\"react-loadable\\";
let LoadableMyComponent = Loadable({
  loader: () => import(\\"./MyComponent\\" /* webpackChunkName: \\"my-component\\" */),
  webpackChunkName: \\"my-component\\",
  webpackRequireWeakId: () => require.resolveWeak(\\"./MyComponent\\"),
  LoadingComponent: MyLoadingComponent
});"
`;

exports[`add-webpack-chunk-name-property 2`] = `
"import Loadable from \\"react-loadable\\";
let LoadableMyComponent = Loadable({
  loader: () => import( /*webpackChunkName: \\"my-component\\"*/\\"./MyComponent\\"),
  webpackChunkName: \\"my-component\\",
  webpackRequireWeakId: () => require.resolveWeak(\\"./MyComponent\\"),
  LoadingComponent: MyLoadingComponent
});"
`;

exports[`not-overwrite-existing-properties 1`] = `
"import Loadable from \\"react-loadable\\";
let LoadableMyComponent = Loadable({
  loader: () => import(\\"./MyComponent\\"),
  LoadingComponent: MyLoadingComponent,
  serverSideRequirePath: null,
  webpackRequireWeakId: null
});"
`;

exports[`track-imports-correctly 1`] = `
"import _path from 'path';
import L from 'react-loadable';
import Loadable from 'not-react-loadable';
let C1 = L({
  loader: () => import('./MyComponent'),
  serverSideRequirePath: _path.join(__dirname, './MyComponent'),
  webpackRequireWeakId: () => require.resolveWeak('./MyComponent')
});
let C2 = Loadable({
  loader: () => import('./MyComponent')
});"
`;
